### PR TITLE
Configure bot: License Header Lint GCF

### PR DESCRIPTION
**Pull request by NimJay**
* See https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/issues/963.
* I have enabled [License Header Lint GCF](https://github.com/apps/license-header-lint-gcf) in this repo.
* This pull-request configures the bot.
* The bot simply runs a GitHub check called `header-check` that warns us about invalid/missing license headers.
* How to review:
    * ensure configuration file looks fine (e.g., list of file extensions)
    * ensure that the bot is actually working inside this pull-request
        * You can see that this bot works in the [commits from this pull-request](https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/pull/964/commits).
        * Look at the commits titled:
            * [Temporarily use incorrect license year](https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/pull/964/commits/b71ff8c0da85440a2dab2f5ecedb24ad03b18b37)
            * [Temporarily remove license header](https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/pull/964/commits/035a79485bd1bcbdd11d09aea02bb61addea85a0)
* This bot does not add the headers for us. It merely warns us of missing/invalid headers.

### CODE REVIEW

File: .github/header-checker-lint.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  - 'java'
+  - 'js'
+  - 'php'
+  - 'py'
+  - 'rb'
+  - 'sh'
+  - 'tf'
+  - 'txt'
+  - 'yaml'
+  - 'yml'
```
Comment (NimJay) [inline .github/header-checker-lint.yml:39]: Let me know if I should remove this chunk:
```
# If you want to ignore checking the license year, uncomment the line below.
# ignoreLicenseYear: true # Useful when migrating in code licensed at previous years.
```
(e.g., if you think it's redundant)

**Review by minherz — APPROVED** [review]
lgtm

nit: did you check that the list of extensions is exhausting?

File: .github/header-checker-lint.yml (hunk 1 above)
Comment (minherz) [reply .github/header-checker-lint.yml:39]: I think `txt` should be removed

Comment (bourgeoisor) [reply .github/header-checker-lint.yml:38]: +1 we don't need copyright headers in e.g. `requirements.txt` files

Comment (NimJay) [reply .github/header-checker-lint.yml:44]: Sounds good!
I've removed both:
```
ignoreFiles:
  - '**/requirements.txt'
```
and
```
sourceFileExtensions:
  - 'txt'
```

**Comment by NimJay** [comment]
@minherz asked:
> did you check that the list of extensions is exhausting?

Initially, no. :)
I just ran `find . -type f -name '*.*' | sed 's|.*\.||' | sort | uniq -c` ([source: Stackoverflow](https://stackoverflow.com/a/31870034)) to check how many of each file extension we have:
```
...
   2 class
   2 cs
   1 csproj
   5 dockerignore
   2 gcloudignore
   7 gitignore
  12 go
   1 hcl
   1 helmignore
   3 html
   1 in
   2 java
   8 js
  19 json
   1 key
   2 lock
  63 md
   7 mod
   2 php
   5 pkl
   1 properties
   2 proto
  19 py
   1 pyi
   1 rb
   3 rev
  21 sh
   3 sql
   7 sum
   1 template
 147 tf
   1 tfstate
   1 tgz
   3 tpl
   9 txt
   2 xml
 227 yaml
  23 yml
```
Some of these **support comments**:
* `.csproj`
* `.dockerignore`
* `Gemfile`
* `.gitignore`
* `.helmignore`
* `Procfile`
* `.properties`
* `.proto`
* `.sql`
* `.tpl`
* `.xml`
* etc...

But there are very few of these files, and I don't think we should support 100% coverage.
I'll add `gitignore` since there are 7 (more than 5) of them, but we should ignore the rest.
Let me know if you think there are some that we _should_ add.

**Comment by bourgeoisor** [comment]
We don't need licenses on things like `.gitignore` or those other mentioned extensions imo

Source / script code + YAML manifests is sufficient.

File: .github/header-checker-lint.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+# ignoreFiles:
+#  - '**/requirements.txt'
+
+# If you want to ignore checking the license year, use ignoreLicenseYear.
+# ignoreLicenseYear: true # Useful when migrating in code licensed at previous years.
+
+sourceFileExtensions:
+  - 'cs'
+  - 'Dockerfile'
+  - 'gitignore'
```
[the remark this answers is not in the document]
Comment (NimJay) [reply .github/header-checker-lint.yml]: Sounds good. I just removed `.gitignore`.

**Review by bourgeoisor — APPROVED** [review]

**Pull request merged**